src/deserialize.rs

```diff
@@ -27,13 +43,55 @@ where
   deserialize(deserializer, validate::aws_secret, "AWS secret key")
 }
 
+pub fn base_uri<'d, D>(deserializer: D) -> Result<String, D::Error>
+where
+  D: Deserializer<'d>,
+{
+  deserialize(deserializer, validate::base_uri, "base URI")
+}
+
 pub fn host<'d, D>(deserializer: D) -> Result<String, D::Error>
 where
   D: Deserializer<'d>,
 {
   deserialize(deserializer, validate::host, "host name or IP address")
 }
 
+pub fn period<'d, D>(deserializer: D) -> Result<u64, D::Error>
+where
+  D: Deserializer<'d>,
+{
+  fn fail<'d, D>(value: &str) -> Result<u64, D::Error>
+  where
+    D: Deserializer<'d>,
+  {
+    Err(D::Error::invalid_value(
+      Unexpected::Str(&value),
+      &"time period",
+    ))
+  }
+
+  let value: String = Deserialize::deserialize(deserializer)?;
+  if let Some(matches) = PERIOD_FORMAT.captures(&value) {
+    if let Ok(multiplier) = matches.get(1).map_or(Ok(1), |m| m.as_str().parse::<u64>()) {
+      match matches.get(2).map_or("", |m| m.as_str()) {
+        "second" => Ok(multiplier * SECOND),
+        "minute" => Ok(multiplier * MINUTE),
+        "hour" => Ok(multiplier * HOUR),
+        "day" => Ok(multiplier * DAY),
+        "week" => Ok(multiplier * WEEK),
+        "month" => Ok(multiplier * MONTH),
+        "year" => Ok(multiplier * YEAR),
+        _ => fail::<D>(&value),
```

Comment:
I think we should add some explicit testcases for this mapping of time strings to integers.